### ISSUE TITLE
Remove simulated data and replace (where used in e.g. testing) with code to simulate

### ISSUE THREAD

**Issue by ben18785** [epiverse-trace/serofoi#160]
Including replacing calls to simulated data with code in vignettes.

**Comment by ben18785** [comment]
@ntorresd I am trying to do this but I don't seem to have sufficient information to determine how one of the simulated data surveys was constructed, specifically: `simdata_sw_dec`. When I look at the documentation for this function, I find:

> Simulated dataset describing a situation where the force-of-infection follows a step-wise decreasing tendency over a period of 50 years (2000-2050) The hypothetical serosurvey is conducted in year 2050 for 250 individuals that are up to 50 years old.

This doesn't say what the FOI is (only that it is stepwise decreasing). Any ideas? The other two simulated datasets have more complete documentation, so they're fine.

**Comment by ntorresd** [comment]
Yes! The FOI in this case is given by:
```
no_transm <- 0.0000000001
foi_sim_sw_dec <- c(rep(0.2, 25), rep(0.1, 10), rep(no_transm, 17))
```
This is the red line in chunk `model_2_plot` in the [foi_models.Rmd vignette](https://github.com/epiverse-trace/serofoi/blob/main/vignettes/foi_models.Rmd).

**Comment by ntorresd** [comment]
This issue is partially addressed by #171 for the time-varying and constant models' tests.
